docs/source/install.rst

```diff
@@ -0,0 +1,316 @@
+(termed 'ARCH's) that specify how PETSc is configured and which external
+packages are built. The currently supported ARCHs are:
+
+* ``default``: the default installation, suitable for most users
+* ``complex``: an installation where PETSc is configured using complex numbers
+
+The different configurations can be selected by passing the flag ``--arch`` to
+``firedrake-configure``. For example::
+
+  $ python3 firedrake-configure --show-system-packages --arch complex
+
+If ``--arch`` is not specified then ``default`` is used.
+
+
+.. _install_system_dependencies:
+
+Installing system dependencies
+------------------------------
+
+If on Ubuntu or macOS, system dependencies can be installed with
+``firedrake-configure``. On Ubuntu run::
+
+  $ sudo apt install $(python3 firedrake-configure --show-system-packages)
+
+which will install the following packages:
+
+.. literalinclude:: apt_deps.txt
+   :language: text
+
+If on macOS you should instead run::
+
+  $ brew install $(python3 firedrake-configure --show-system-packages)
+
+which will install the following packages:
+
+.. literalinclude:: homebrew_deps.txt
+   :language: text
+
+If you do not have one of these systems then these dependencies will need to
+be installed manually.
+
+.. note::
+   Not all the system dependencies declared by ``firedrake-configure`` have to
+   be installed at this stage. Some (e.g. HDF5, hwloc) can also be installed
+   from source by PETSc during the ``configure`` :ref:`step<install_petsc>` by
+   passing additional flags (e.g. ``--download-hdf5``, ``--download-hwloc``).
+
+
+.. _install_petsc:
+
+Installing PETSc
+----------------
+
+For Firedrake to work as expected, it is important that a specific version of PETSc_
+is installed with a specific set of external packages. To install PETSc you need to
+do the following steps:
+
+#. Clone the PETSc repository::
+
+   $ git clone https://github.com/firedrakeproject/petsc.git
+   $ cd petsc
+
+#. Run PETSc ``configure``, passing in the flags generated by ``firedrake-configure``::
+
+   $ ./configure $(python3 ../firedrake-configure --show-petsc-configure-options)
+
+#. Compile PETSc::
+
+   $ make
+
+If you are using one of the
+:ref:`officially supported distributions<supported_systems>` then these configure
+options will include paths to system packages so PETSc can correctly find and
+link against them. If you are not then you should pass the ``--no-package-manager``
+flag to obtain a set of configure options where ``firedrake-configure``
+pessimistically assumes that no external packages are available, and hence need
+to be downloaded and compiled from source::
+
+   $ ./configure $(python3 ../firedrake-configure --no-package-manager --show-petsc-configure-options)
+
+For the ``default`` ARCH, running ``firedrake-configure`` with
+``--no-package-manager`` will produce the flags:
+
+.. literalinclude:: petsc_configure_options.txt
+   :language: text
+
```

Comment:
Is this missing the `make all` call after configuring?

Reply:
And possibly a recommendation to run `make check` afterwards to make sure the PETSc installation has worked before moving on

Reply:
I don't think I need to say to `make` here because I am specifically talking about modifying the call to `./configure`. I'll mention `make check`.

Reply:
Don't you need to actually build PETSc after configuring it though?

Reply:
But I do [here](https://github.com/firedrakeproject/firedrake/pull/4011/files/172244b5f80a6e2773676f96d0d77eeef92331a1#diff-3b2a6fca7ca7832ce28a07b57db259a44a6d6587270e3e576f53e32a6553ed05R130)?

Reply:
You are completely correct. I think I had somehow opened up an old version of this file that wasn't showing it in the preview. Ignore this.